### ISSUE TITLE
User Sign-In/log-in

### ISSUE THREAD

**Issue by MiguelGibo** [uprm-inso4101-2024-2025-s1/semester-project-campus-resource-and-event-management#12]
**Objective:**
Implement the availability for a user to sign-up/log-in into their account.

**Summary:** 
To manage reservations and listings of event spaces, users need the ability to create accounts and sign-up/log-in. Allowing users to easily manage their reservations and event listings.

**Proposed Solution:** 
- a user interface that prompts user to log-in.
- If a user doesn't have an account, it prompts user to sign-up instead. Which allows users to sign-up with email, password, name, and student ID.
- If user already has an account, open user profile.
- Display error message if either the log-in or sign-up process failed, such as incorrect email or password.
- Once logged-in into their profile where they manage their event listing or reservations.

**Comment by Ojani** [comment]
Since there isn't any authentication yet present. Should the login and sign up just be dummy forms that lead you to a placeholder profile?

**Comment by MiguelGibo** [comment]
The sign-up should create a new user. 
The login should allow access to already existing users, it should link to the homepage of said user or their user profile. 
In retrospect, those seem like separate but related tasks, so it would be wise to split them up into separate tickets.
In addition, I will also be creating a new task for implementing the user profile itself.

@AngelUPR Would it be possible to replace this task with 2 separate tasks. 
One to implement the sign-up and user enrollment,
and one to implement a login and user authorization.

**Comment by AngelUPR** [comment]
> The sign-up should create a new user. The login should allow access to already existing users, it should link to the homepage of said user or their user profile. In retrospect, those seem like separate but related tasks, so it would be wise to split them up into separate tickets. In addition, I will also be creating a new task for implementing the user profile itself.
> 
> @AngelUPR Would it be possible to replace this task with 2 separate tasks. One to implement the sign-up and user enrollment, and one to implement a login and user authorization.

Absolutely. Please start a detailed description in discussion about how exactly you want to split them up and the implementation so I can assign that discussion as an issue
